source-container-build/app/test_source_build.py

```diff
@@ -1292,3 +1292,28 @@ def test_source_image_does_not_exist(self, mock_run: MagicMock):
         source_image = source_build.resolve_source_image_by_manifest("registry.io:3000/ns/app:1.0")
 
         self.assertIsNone(source_image)
+
+
+class TestUnsignedSourceImage(unittest.TestCase):
+
+    def setUp(self):
+        self.called_proc_err = CalledProcessError(
+            -1,
+            "mock cmd",
+            stderr="Source image rejected: A signature was required, but no signature exists".encode(),  # noqa: E501 (ignore line too long)
+        )
+
+    @patch("source_build.run")
+    def test_source_image_is_unsigned_and_not_ingored(self, run: MagicMock):
+        run.side_effect = self.called_proc_err
+
+        with pytest.raises(source_build.NoSignatureError):
+            source_build.download_parent_image_sources("", "", False)
```

Comment:
nitpick: this would be much more obvious at first sight
```suggestion
            source_build.download_parent_image_sources("", "", ignore_unsigned_image=False)
```

Reply:
Thanks for the suggestion!